include/rcpputils/filesystem_helper.hpp

```diff
@@ -511,7 +511,7 @@ inline bool create_directories(const path & p)
 #endif
     }
   }
-  return status == 0;
+  return status == 0 && p_built.is_directory();
```

Comment:
Small change, but I switched these to rely on short-circuiting to avoid calling `is_directory()` when it's not necessary